### ISSUE TITLE
What is S rank?

### ISSUE THREAD

**Issue by hugovk** [anuraghazra/github-readme-stats#330]
**Describe the bug**

When I add `include_all_commits=true` my badge goes from A++ to S+

**Expected behavior**

I don't know what I expected, but I've no idea what S+ means. S is way down the alphabet, is it much worse than A-F?

**Screenshots / Live demo link (paste the github-readme-stats link as markdown image)**

```md
[![hugovk's GitHub stats](https://github-readme-stats.vercel.app/api?username=hugovk&count_private=true&show_icons=true&hide_title=true)](https://github.com/anuraghazra/github-readme-stats)
```
[![hugovk's GitHub stats](https://github-readme-stats.vercel.app/api?username=hugovk&count_private=true&show_icons=true&hide_title=true)](https://github.com/anuraghazra/github-readme-stats)

Add `include_all_commits=true`:

```md
[![hugovk's GitHub stats](https://github-readme-stats.vercel.app/api?username=hugovk&count_private=true&show_icons=true&hide_title=true&include_all_commits=true)](https://github.com/anuraghazra/github-readme-stats)
```
[![hugovk's GitHub stats](https://github-readme-stats.vercel.app/api?username=hugovk&count_private=true&show_icons=true&hide_title=true&include_all_commits=true)](https://github.com/anuraghazra/github-readme-stats)

**Screenshot**

![image](https://user-images.githubusercontent.com/1324225/89691122-cf5e7680-d910-11ea-8157-167fefc6a35d.png)

**Comment by jopemachine** [comment]
> I don't know what I expected, but I've no idea what S+ means. S is way down the alphabet, is it much worse than A-F?

I think S means much better than A or A++. (not much worse than A-F).

I think the rating has gone up because the total number of commit is much higher when you include all your commit (private commits).

**Comment by Metalhearf** [comment]
This maybe could interest you, about the _S to F_ grading.
https://en.wikipedia.org/wiki/Academic_grading_in_Japan

**Comment by hugovk** [comment]
OK, thanks, good to know!

I wonder how widely know Japanese grading is. Perhaps it'd be better to shift the grades to begin with A and end with, say, C.

Or add some extra pluse to the existing to A, rather than S.

Thanks!

**Comment by anuraghazra** [comment]
Yeah I guess I'll change this. A lot of people recommend me not to use S because it's confusing. 

I intentionally did this to curve the rankings toward A and A++ even for low active users.

**Comment by avelino** [comment]
> This maybe could interest you, about the _S to F_ grading.
> https://en.wikipedia.org/wiki/Academic_grading_in_Japan

Could you put in the documentation `README`

**Comment by 7ma7X** [comment]
To be honest, rank S is not so popular even in Japan.
I've lived in Japan for 20 years and I've never had such grading systems.

**Comment by anuraghazra** [comment]
> I've lived in Japan for 20 years and never had such grading systems.

It's mostly used in fictional grading. highly used in modern PC games.

**Comment by landonepps** [comment]
> I intentionally did this to curve the rankings toward A and A++ even for low active users.

Is it even possible to get a B+?
https://github-readme-stats.vercel.app/api?username=tranminhson100
Here's a user with nearly as little activity as possible, and they have an A+.

**Comment by AryanBeezadhur** [comment]
I'm disappointed to know that I have the same ranking as [this user](https://github-readme-stats.vercel.app/api?username=tranminhson100) when I am [much more active](https://github-readme-stats.vercel.app/api?username=aryanbeezadhur&count_private=true&include_all_commits=true) :(

![https://github-readme-stats.vercel.app/api?username=tranminhson100](https://github-readme-stats.vercel.app/api?username=tranminhson100)

![https://github-readme-stats.vercel.app/api?username=aryanbeezadhur&count_private=true&include_all_commits=true](https://github-readme-stats.vercel.app/api?username=aryanbeezadhur&count_private=true&include_all_commits=true)

**Comment by AryanBeezadhur** [comment]
I think rank is largely biased towards PRs and issues, not commits, because look at this user with a top rank and no commits:

![https://github-readme-stats.vercel.app/api?username=vscode-triage-bot&show_icons=true&count_private=true](https://github-readme-stats.vercel.app/api?username=vscode-triage-bot&show_icons=true&count_private=true)

**Comment by HelloRusk** [comment]
https://github.com/anuraghazra/github-readme-stats/blob/master/src/calculateRank.js#L28-L34

The number of commits or contributions is weighted only 1.65 times as the number of issues you've opened. This evaluation is somewhat mismatched.
Maybe we should open another issue to discuss better way of ranking.

**Comment by HelloRusk** [comment]
related to #455

**Comment by hugovk** [comment]
Stale bot, please keep open, awaiting changes: https://github.com/anuraghazra/github-readme-stats/issues/330#issuecomment-671012692

**Comment by hugovk** [comment]
stalebot, still relevant, thanks!

**Comment by hugovk** [comment]
stalebot, still relevant, thanks!

**Comment by anuraghazra** [comment]
Hi @hugovk I think your original question is being answered that S+ Rank is better than A++ 

And for the rank evaluation we are already discussing it in #455

Closing this for now as resolved. feel free to open another issue if you have any other questions.

**Comment by hugovk** [comment]
Oh, I thought you were going to change it:

> Yeah I guess I'll change this. A lot of people recommend me not to use S because it's confusing.
> 
> I intentionally did this to curve the rankings toward A and A++ even for low active users.

If not, I'll probably remove the ranking from my README, thank you anyway for your work!

**Comment by tobealive** [comment]
S Tier is a common thing in terms of ranking. I like it 👍

**Comment by Cry199** [comment]
It would be cool after S++ to have a GOD mode

**Comment by peterviegas** [comment]
What does the letter C mean?

**Comment by qwerty541** [comment]
Hey @peterviegas 

From the documentation:

Available ranks are S (top 1%), A+ (12.5%), A (25%), A- (37.5%), B+ (50%), B (62.5%), B- (75%), C+ (87.5%) and C (everyone). This ranking scheme is based on the [Japanese academic grading](https://wikipedia.org/wiki/Academic_grading_in_Japan) system. The global percentile is calculated as a weighted sum of percentiles for each statistic (number of commits, pull requests, reviews, issues, stars, and followers), based on the cumulative distribution function of the [exponential](https://wikipedia.org/wiki/exponential_distribution) and the [log-normal](https://wikipedia.org/wiki/Log-normal_distribution) distributions. The implementation can be investigated at [src/calculateRank.js](https://github.com/anuraghazra/github-readme-stats/blob/master/src/calculateRank.js). The circle around the rank shows 100 minus the global percentile.
